server/src/main/java/io/crate/replication/logical/LogicalReplicationService.java

```diff
@@ -366,18 +376,21 @@ private void verifyTablesDoNotExist(String subscriptionName, PublicationsStateAc
                 subscriptionName,
                 relation
             );
-            consumer.accept(new RelationAlreadyExists(relation, message));
+            onFailure.accept(new RelationAlreadyExists(relation, message));
         };
         for (var index : stateResponse.concreteIndices()) {
             if (metadata.hasIndex(index)) {
                 onExists.accept(RelationName.fromIndexName(index));
+                return;
```

Comment:
Wouldn't that `return` make the rest of the method skip ?

Reply:
Yes, it exists the method.